addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -309,6 +313,59 @@ var Wysiwyg = Widget.extend({
             mediaDialog.on('cancel', this, resolve);
         });
     },
+    changeSelection(commandId, color, $dropDownToToggle = undefined) {
+        if(color === "") {
+            this.editor.execCommand('un' + commandId); // todo : not ideal : use a map to get cammands from actions ?
+        } else {
+            if (this.colorPickers[commandId].colorNames.indexOf(color) !== -1) {
+                // todo : find a better way to detect and send css variable
+                color = "var(--" + color + ")";
+            }
+            this.editor.execCommand(commandId, {color: color});
+        }
+        if($dropDownToToggle !== undefined) {
+            $dropDownToToggle.find(".dropdown-toggle").dropdown("toggle");
+        }
+    },
+    initColorPicker(colorPickerCommandId, $dropdownNode) {
+        return new Promise(()=> {
+            if (!this.colorPickers[colorPickerCommandId]) {
+                // Init the colorPalete for this color picker Dropdown.
+                this.colorPickers[colorPickerCommandId] = new ColorPaletteWidget(this, {});
+
+                // The <form> prevent the dropdown to automaticaly close when user click inside.
+                const $form = $('<form >');
+                $form.on("submit", function(e) {
+                    e.stopImmediatePropagation();
+                    e.preventDefault();
+                });
```

Comment:
The problem should be corrected at the source. I see in odoo master that it's the "normal" behavior of the color picker in a modal. If the behavior is to not close, it should be fixed in the color picker itself.

I would not try to correct that in this commit.